### DISCUSSION TITLE
Add and remove multi-line line comments

### DISCUSSION THREAD

**Discussion by bbarker** [helix-editor/helix#8883] [Q&A] [answered]
I'm pretty new to helix so apologies for likely missing something obvious, and it's ok by me if this ability doesn't exist yet, but wanted to check:

Say i have some code:

```rust
fn json_pointer_to_value<'a>(json: &'a Value, pointer: &str) -> Option<&'a Value> {
    pointer.split('.').try_fold(json, |acc, part| match acc {
        Value::Object(map) => map.get(part),
        _ => None,
    })
}
```

and I want to comment out some lines so it looks like

```rust
fn json_pointer_to_value<'a>(json: &'a Value, pointer: &str) -> Option<&'a Value> {
    pointer.split('.').try_fold(json, |acc, part| match acc {
        // Value::Object(map) => map.get(part),
        // _ => None,
    })
}
```

Then i'd like to be able to select those lines and get back to the original (uncommented) state. What are the ways one can achieve this presently, if any?

**Comment by David-Else** [comment]
Select them both (maybe use `x`) and press the toggle comment shortcut:

```
[keys.normal]
"#" = "toggle_comments"
```
My brain forgot the default on purpose as it was not good :)

You can select both and press it again and hey presto they are uncommented!

There is PR to do proper multi line block comments in the works: https://github.com/helix-editor/helix/pull/4718

**Reply by gabydd** [reply, marked as answer]
The current default keybinding is ``C-c``

**Reply by David-Else** [reply]
Ctrl-C sends an interrupt signal (SIGINT) to the current process running in the foreground in the terminal, no wonder I rejected outright the idea of using that for toggling comments... 99% of decisions in Helix are perfect, but that one was goofy as hell. I think the new PR will have it as space c which will be fine :)

**Reply by pascalkuthe** [reply]
the idea is based on vim (the shortcut is pretty universal among comment plugins there). I do personally prefer it on `pound` (#)

**Comment by themixednuts** [comment]
Is there actual multiline comment support and not multiple single line comments?

As well as something of the equivalent for Svelte files and such (https://github.com/JoosepAlviste/nvim-ts-context-commentstring)

**Reply by kirawi** [reply]
See the PR linked above https://github.com/helix-editor/helix/discussions/8883#discussioncomment-7646172
